src/main/java/org/aksw/simba/lemming/creation/Inferer.java

```diff
@@ -198,15 +217,15 @@ public void iterateStmts(Model newModel, Model sourceModel, OntModel ontModel, M
 			List<? extends OntResource> domain = property.listDomain().toList();
 			for (OntResource curResource : domain) {
 				Statement subjType = ResourceFactory.createStatement(subject, RDF.type, curResource);
-				if (!newModel.contains(subjType)) {
+				if (!model.contains(subjType)) {
 					newStmts.add(subjType);
 				}
 			}
 			if (object.isResource()) {
 				List<? extends OntResource> range = property.listRange().toList();
 				for (OntResource curResource : range) {
 					Statement objType = ResourceFactory.createStatement(object.asResource(), RDF.type, curResource);
-					if (!newModel.contains(objType)) {
+					if (!model.contains(objType)) {

```

Comment:
Same thing as the comment regarding `model.contains(stmt);`

Reply:
Done